src/features/CareViewPatients/components/CareViewPatients.jsx

```diff
@@ -54,10 +55,20 @@ export const CareViewPatients = () => {
     previous: false,
     next: false,
   });
-  const currentShiftArray =
-    ipdConfig.shiftDetails &&
-    currentShiftHoursArray(new Date(), ipdConfig.shiftDetails)
-      .currentShiftHoursArray;
+  const { currentShiftArray, rangeArray, shiftIndex } = useMemo(() => {
+    return {
+      currentShiftArray:
+        ipdConfig.shiftDetails &&
+        currentShiftHoursArray(new Date(), ipdConfig.shiftDetails)
```

Comment:
we could just store the returned object from currentShiftHoursArray method in a variable and then assign currentShiftArray, rangeArray and shiftIndex values. This way the currentShiftHoursArray method won't be invoked 3 times and only once